x/subscription/keeper/subscription.go

```diff
@@ -316,10 +324,10 @@ func (k Keeper) advanceMonth(ctx sdk.Context, subkey []byte) {
 			sub.MonthCuTotal = plan.PlanPolicy.TotalCuLimit
 
 			k.resetSubscriptionDetailsAndAppendEntry(ctx, &sub, block, false)
-		} else if sub.AutoRenewal {
+		} else if sub.AutoRenewalNextPlan != types.AUTO_RENEWAL_PLAN_NONE {
```

Comment:
can you move this to the subscription object so we can call:
sub.AutoRenewal()?